darts/models/torch_forecasting_model.py

```diff
@@ -616,18 +616,37 @@ def predict_from_dataset(self,
 
                 # repeat prediction procedure for every needed sample
                 batch_predictions = []
-                for i in range(num_samples):
+
+                # if batch_size is not a multiple of input_series, this won't fill the entire batch_size
+                series_length = input_series.shape[0]
+                batch_sample_size = max(batch_size // series_length, 1)
```

Comment:
We might have to set `batch_sample_size` equal to 1 explicitly when `num_samples==1`. This is because we could have a `batch_size > 1` to speed up multiple deterministic predictions. But in the last batch, it might be that `series_length < batch_size`. Then we can have a `batch_sample_size` that is bigger than 1, which will lead to tiling, even though we don't want to make a probabilistic forecast

Reply:
I don't believe this occurs, as the loop will check whether sample_length+batch_size > num_samples. So it will never generate more samples than specified. However I added min around batch_sample_size to make it more explicit.